service/controller/v1/prometheus/service.go

```diff
@@ -152,7 +145,11 @@ func (s *Service) isReloadRequired(ctx context.Context) (bool, error) {
 		return false, nil
 	}
 
-	if s.IsReloadRequested() {
+	s.isReloadRequestedMutex.Lock()
+	isReloadRequested := s.isReloadRequested
+	s.isReloadRequestedMutex.Unlock()
```

Comment:
I'd like to get rid of `isReloadRequested` that completely. I'll see if this is possible later on.